File: lat2/2.py
import pandas as pd
import mysql.connector
import matplotlib.pyplot as plt

connection = mysql.connector.connect(host='localhost', user='root', password='', database='pembelian_shopee')
query1 = "SELECT*FROM order_detail"
query2 = "SELECT*FROM sku_detail"
df1 = pd.read_sql(query1, connection)
df2 = pd.read_sql(query2, connection)
connection.close()

df = pd.merge(df1, df2, left_on='sku_id', right_on='id', how='inner')
df = df[df['is_valid'] == 1]
df['order_date'] = pd.to_datetime(df['order_date'])
df = df[df['order_date'].dt.year == 2022]
tabel1 = df.groupby(['sku_name', 'category'])[['after_discount', 'qty_ordered']].sum().reset_index()
tabel1 = tabel1.sort_values('qty_ordered', ascending=False).reset_index(drop=True).head(10)
print(tabel1)

tabel2 = df.drop_duplicates('customer_id')
tabel2 = tabel2.groupby(['sku_name', 'category'])['customer_id'].nunique().reset_index(name='jumlah pelanggan')

print(tabel2)
tabel3 = df.drop_duplicates('id_x')
tabel3 = tabel3.groupby(['sku_name', 'category'])['id_x'].nunique().reset_index(name='total order')
print(tabel3)
tabel_awal = pd.merge(tabel1, tabel2, on=['sku_name', 'category'], how='inner')
tabel_akhir = pd.merge(tabel_awal, tabel3, on=['sku_name', 'category'], how='inner')
tabel_akhir = tabel_akhir.sort_values('qty_ordered', ascending=False).reset_index(drop=True).head(10)
print(tabel_akhir)

File: lat2/2.r
library(RMySQL)
library(dplyr)
library(lubridate)

# Ganti 'dbname', 'username', 'password', dan 'host' sesuai dengan informasi database Anda
con <- dbConnect(MySQL(), dbname = "pembelian_shopee", user = "root", password = "", host = "localhost")

query1 <- "SELECT * FROM order_detail"
df1 <- dbGetQuery(con, query1)
query2 <- "SELECT * FROM sku_detail"
df2 <- dbGetQuery(con, query2)
dbDisconnect(con)

# df <- merge(df1, df2, by.x = "sku_id", by.y = "id")
df <- inner_join(df1, df2, by = c("sku_id" = "id"))
print(head(df))

df <- df[df$is_valid == 1, ]
df$order_date <- as.Date(df$order_date)
df <- filter(df, year(order_date) == 2022)

tabel1 <- df %>%
    group_by(sku_name, category) %>%
    summarise(after_discount = sum(after_discount), qty_ordered = sum(qty_ordered)) %>%
    arrange(desc(qty_ordered))
tabel1 <- head(tabel1, 10)
print(tabel1)


tabel2 <- df %>% distinct(customer_id, .keep_all = TRUE) %>% group_by(sku_name, category) %>% summarise(jumlah_pelanggan = n_distinct(customer_id))
print(head(tabel2))



tabel3 <- df %>% distinct(id, .keep_all = TRUE) %>% group_by(sku_name, category) %>% summarise(total_order = n_distinct(id))
print(head(tabel3))
cat('\n\n\n\n\n\n')


tabelawal <- merge(tabel1, tabel2, by = c('sku_name', 'category'))
tabelakhir <- merge(tabelawal, tabel3, by = c('sku_name', 'category'))
tabelakhir <- tabelakhir %>% arrange(desc(qty_ordered))
print(tabelakhir)
